### PR TITLE
Various QoL MapEditor changes and issue 1856

### LINKED ISSUE #1856: 47.5 (and earlier) Map Editor - Bridges not appearing on preview hex
In v0.47.5 (and previous versions), when a bridge has been added to a hex, the bridge does not appear in the hex terrain preview hex.
Not sure whether this is unique to me, or a general issue to do with how bridges are displayed.

### PR DESCRIPTION
**Pull request by SJuliez**
- Several MapEd file dialogs (save, load etc.) now, finally, remember their size.
- The validation dialogs now appear centered and react to ESC or Enter.
- The board validation upon loading a board now does show the error report because it can be different from what a later validation shows, especially when buildings have errors.
- The minimap (in the map editor) now shows invalid hexes as bright red.
- The MapEd frame title now shows if the board has been changed (with the usual "*"). A fresh random board does not count as changed.
- If the board is changed, the MapEd will now not just close when the window "X" is pressed but ask if the user wants to save the board first.
- The RandomMap dialog at the startup of the MapEd can now be disabled in the dialog.
- Bridges and other orthos now show up in the work hex - Resolves #1856

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/BoardEditor.java [hunk 1]
```diff
@@ -1345,16 +1376,19 @@ private void updateWhenSelected() {
...
+        if (!userCancel) {
+            board = BoardUtilities.generateRandom(mapSettings);
+            game.setBoard(board);
+            curfile = null;
+            butSourceFile.setEnabled(false);
+            frame.setTitle(Messages.getString("BoardEditor.title")); //$NON-NLS-1$
+            menuBar.setBoard(true);
     		bvc.doLayout();
     		resetUndo();
     		choTheme.setSelectedItem(mapSettings.getTheme());
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/BoardEditor.java:1394]: I think this is tab versus space spacing, which would be good to fix.

File: megamek/src/megamek/client/ui/swing/BoardEditor.java [hunk 2]
```diff
@@ -1759,21 +1807,17 @@ public void actionPerformed(ActionEvent ae) {
                 try {
                     Desktop.getDesktop().open(curfile);
                 } catch (IOException e) {
-                    JOptionPane.showMessageDialog(this, "Could not open the file "+curfile+". "+e.getMessage());
+                    ignoreHotKeys = true;
+                    JOptionPane.showMessageDialog(frame, "Could not open the file "+curfile+". "+e.getMessage());
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/BoardEditor.java:1811]:
```suggestion
                    JOptionPane.showMessageDialog(frame, "Could not open the file " + curfile + ". " + e.getMessage());
```

File: megamek/src/megamek/client/ui/swing/BoardEditor.java [hunk 3]
```diff
@@ -2069,6 +2124,31 @@ public boolean shouldIgnoreHotKeys() {
...
+    
+    /** 
+     *  Sets the Board Editor frame title, adding the current file name if any
+     *  and a "*" if the board has been changed.
+     */
+    private void setFrameTitle() {
+        String title = Messages.getString("BoardEditor.title"); //$NON-NLS-1$
+        if (curfile != null) {
+            title = Messages.getString("BoardEditor.title") + curfile; //$NON-NLS-1$
+        }
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/BoardEditor.java:2146]:
```suggestion
        String title = Messages.getString("BoardEditor.title"); //$NON-NLS-1$
        if (curfile != null) {
            title += curfile;
        }
```

File: megamek/src/megamek/client/ui/swing/MiniMap.java [hunk 4]
```diff
@@ -728,6 +728,13 @@ protected synchronized void drawMapOrig() {
                         paintCoord(gg, j, k, true);
                     }
                     addRoadElements(h, j, k);
+                    // Color invalid hexes red when in the Map Editor
+                    if ((m_game != null) && 
+                            (m_game.getPhase() == IGame.Phase.PHASE_UNKNOWN)
+                            && !h.isValid(null)) {
+                        gg.setColor(Color.RED);
+                        paintCoord(gg, j, k, true);
+                    }
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/MiniMap.java:737]: Is red really the best choice, given that some people can't see it?

File: megamek/src/megamek/client/ui/swing/RandomMapDialog.java [hunk 5]
```diff
@@ -271,8 +274,22 @@ private JPanel setupTopPanel() {
...
+            showAtStartButton.addActionListener(this);
+            showAtStartButton.setMnemonic(showAtStartButton.getText().charAt(0));
+            showAtStartButton.setSelected(guip.getBoardEdRndStart());
+            leftPanel.add(showAtStartButton);
+        }
+        
+        // The main panel with the Okay, Cancel etc. buttons
         JPanel panel = new JPanel(new FlowLayout(FlowLayout.CENTER, 2, 2));
-
+
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/RandomMapDialog.java:292]:
```suggestion

```

**Comment by SJuliez** [comment]
Thanks for checking, @Windchild292 
I corrected some tab spacings, updated the Message system to the more useful (I think) Object... instead of Object[] parameter (can save some "new Object[] {...}"). Also removed a few one-line ifs which were my own wrong style originally. (I just recently noticed there is a coding style guide in the wiki)
Concerning Color.Red and visibility, red is still the usual warning/bad color everywhere, MegaMek or otherwise. I have no problem using GUIPreferences.getWarningColor() or something if it exists but I'd rather not arbitrarily start using Blue now (or whatever would be a good color, I actually don't know even that).

File: megamek/src/megamek/client/ui/swing/BoardEditor.java [hunk 6]
```diff
@@ -1751,6 +1761,17 @@ private void showSettings() {
         // Show the settings dialog.
         setdlg.setVisible(true);
     }
+    
+    private void freshBoardSettings() {
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/BoardEditor.java:1765]: I'm not sure about this name, as it took me a second to realise what it meant.

File: megamek/src/megamek/client/ui/Messages.java [hunk 7]
```diff
@@ -1,16 +1,16 @@
...
- *  Software Foundation; either version 2 of the License, or (at your option)
- *  any later version.
- *
- *  This program is distributed in the hope that it will be useful, but
- *  WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
- *  or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
- *  for more details.
- */
+/*  
+* MegaMek - Copyright (C) 2020 - The MegaMek Team
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/Messages.java:2]: Why are you removing the original side of things?

**Comment by Windchild292** [comment]
If we don't have a getWarningColour, it is a good addition for accessibility (~8.5% of the population is red-green colourblind). Maybe just put a getter in that just returns red, and I'll open up a story to add that for accessibility?

File: megamek/src/megamek/client/ui/swing/BoardEditor.java (hunk 6 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/BoardEditor.java:1765]: Do you have a suggestion? Its not necessarily a new board, not necessarily a loaded board. updateUiForFreshBoard...

File: megamek/src/megamek/client/ui/Messages.java (hunk 7 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/Messages.java:2]: Please explain. Im not remotely an expert on copyright in software applications so I thought why not update it to the current example in the wiki.

**Comment by SJuliez** [comment]
Sounds good. I'll do that.

File: megamek/src/megamek/client/ui/Messages.java (hunk 7 above)
Comment (Windchild292) [reply megamek/src/megamek/client/ui/Messages.java:2]: I did that originally, but was told to instead leave the original and add the new line below it when I first started contributing. Hence the question.

File: megamek/src/megamek/client/ui/swing/BoardEditor.java (hunk 6 above)
Comment (Windchild292) [reply megamek/src/megamek/client/ui/swing/BoardEditor.java:1765]: refreshBoardSettings? Not super fitting, but better. refreshBoardUI?

File: megamek/src/megamek/client/ui/Messages.java (hunk 7 above)
Comment (sixlettervariables) [reply megamek/src/megamek/client/ui/Messages.java:2]: Basically we can't erase Ben's copyright on his work. Unless this is a total rewrite.

**Comment by SJuliez** [comment]
OK, changed the copyright notice and method name and added an accessibility oriented WARNING_COLOR to the GUIP that defaults to red and currently has no UI for changes.

File: megamek/src/megamek/client/ui/swing/BoardEditor.java [hunk 8]
```diff
@@ -2137,17 +2143,25 @@ public void paintComponent(Graphics g) {
                     g.drawImage(newVar, 0, 0, this);
                 }
                 // add level and INVALID if necessary
+                if (guip.getAntiAliasing()) {
+                    ((Graphics2D)g).setRenderingHint(
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/BoardEditor.java:2147]:
```suggestion
                    ((Graphics2D) g).setRenderingHint(
```

**Pull request merged**